src/Event.re

```diff
@@ -1,17 +1,17 @@
 /* Event.re */
 
-type cb('a) = 'a => unit;
+type unsubscribe = unit => unit;
 
-type t('a) = ref(list(cb('a)));
+type t('a) = ref(list('a => unit));
 
 let create = () => ref([]);
 
-let subscribe = (evt: t('a), f: cb('a)) => {
-  evt := List.append(evt^, [f]);
-  let unsubscribe = () => {
-    evt := List.filter(f => f !== f, evt^);
```

Comment:
This was a bug. It didn't remove one callback, but cleared the list entirely `f !== f` is always `false`. That's one argument against shadowing, although I'm surprised this didn't cause an unused warning.